src/routes/error.ts

```diff
@@ -1,55 +1,114 @@
 // Copyright © 2022 Ory Corp
 // SPDX-License-Identifier: Apache-2.0
+import { FlowError, FrontendApi, GenericError } from "@ory/client"
 import { UserErrorCard } from "@ory/elements-markup"
-import { AxiosError } from "axios"
+import { isAxiosError } from "axios"
 import {
-  defaultConfig,
-  isQuerySet,
   RouteCreator,
   RouteRegistrator,
+  defaultConfig,
+  isQuerySet,
 } from "../pkg"
 
+type OAuth2Error = {
+  error: string
+  error_description?: string
+  error_hint?: string
+}
+
+function isOAuth2Error(query: qs.ParsedQs): query is OAuth2Error {
+  return query.error !== undefined
+}
+
+/**
+ * Returns an error object, either from Ory Identities or an OAuth2 error.
+ *
+ * @param frontend the frontend SDK
+ * @param query the query parameters as received from the request
+ * @returns a FlowError object
+ */
+async function fetchError(
+  frontend: FrontendApi,
+  query: qs.ParsedQs,
+): Promise<FlowError> {
+  // If the error is an OAuth2 error, its details are encoded into the URL.
+  // We can simply decode them and return them here.
+  if (isOAuth2Error(query)) {
+    return {
+      id: decodeURIComponent(query.error.toString()),
+      error: {
+        status: "OAuth2 Error",
+        id: decodeURIComponent(query.error.toString()),
+        message: decodeURIComponent(
+          query.error_description?.toString() || "No description provided",
+        ),
+        ...(query.error_hint
+          ? { hint: decodeURIComponent(query.error_hint.toString()) }
+          : {}),
+        code: 599, // Dummy code to trigger the full error screen
```

Comment:
why do we need a dummy code? why not just make it 400?

Reply:
>to trigger the full error screen

Otherwise, we only show the `message`, bot the `id`, etc.

https://github.com/ory/elements/blob/main/src/react-components/ory/user-error-card.tsx#L37

Also, this is most likely a 5XX error, as there is likely a misconfiguration.